libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceSDKManagerTests.m

```diff
@@ -98,50 +116,91 @@ - (void)testOneLaunchAtATime
 {
     [self createStandardPostLaunchBlock];
     [self createTestAppIdentity];
-    
     [self launchAndVerify:YES failMessage:@"Initial launch attempt should have been successful."];
     [self launchAndVerify:NO failMessage:@"Second concurrent launch should not be allowed."];
+    [self verifyPostLaunchState];
+}
+
+- (void)testOverrideAiltnAppNameBeforeSDKManagerLaunch
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [SalesforceSDKManager setAiltnAppName:kTestAppName];
+    [self createStandardPostLaunchBlock];
+    [self createTestAppIdentity];
+    [self launchAndVerify:YES failMessage:@"Launch attempt should have been successful."];
+    [self verifyPostLaunchState];
+    [self compareAiltnAppNames:kTestAppName];
+    [SalesforceSDKManager setAiltnAppName:prevName];
+}
+
+- (void)testOverrideAiltnAppNameAfterSDKManagerInit
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [self createStandardPostLaunchBlock];
+    [self createTestAppIdentity];
+    [self launchAndVerify:YES failMessage:@"Launch attempt should have been successful."];
+    [self verifyPostLaunchState];
+    [SalesforceSDKManager setAiltnAppName:kTestAppName];
+    [self compareAiltnAppNames:kTestAppName];
+    [SalesforceSDKManager setAiltnAppName:prevName];
+}
+
+- (void)testDefaultAiltnAppName
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [self createStandardPostLaunchBlock];
+    [self createTestAppIdentity];
+    [self launchAndVerify:YES failMessage:@"Launch attempt should have been successful."];
+    [self verifyPostLaunchState];
+    NSString *appName = [[NSBundle mainBundle] infoDictionary][(NSString *) kCFBundleNameKey];
+    [self compareAiltnAppNames:appName];
+    [SalesforceSDKManager setAiltnAppName:prevName];
     
+}
+
+- (void)testOverrideInvalidAiltnAppName
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [SFUserAccountManager sharedInstance].currentUser = [self createUserAccount];
```

Comment:
This should also make allowances for the previous current user value, and reset it at the end of the test.